### ISSUE TITLE
Typescript Compilation Errors

### ISSUE THREAD

**Issue by agokhale1** [saul/demofile#154]
**Describe the bug**
Compiling Typescript files that utilize the demofile library results in errors. Specifically, the compiler is unable to find `@types/lodash` and `@types/bytebuffer` and complains about the `EventEmitter` import in `entities.d.ts`.

The compiler is configured as such:
```
{
    "compilerOptions": {
        "target": "es2017",
        "module": "commonjs",
        "allowJs": true,
        "declaration": true,
        "strict": true,
        "outDir": "./build"
    },
    "include": ["./src/**/*.ts"],
    "exclude": ["./demos"]
}
```

**Code to reproduce**
1. Create an empty project
2. Install typescript and demofile
3. Write a simple parser (i.e. print `gameRules.roundsPlayed` on `round_end`)
4. Compilation will result in 6 errors

**Expected behaviour**
There should be no errors during compilation.

**Affected demos**
N/A

**Comment by agokhale1** [comment]
A workaround I found is to install the `@types/lodash` and `@types/bytebuffer` dependencies and modify the `EventEmitter` import:
`import EventEmitter = require("events");` --> `import { EventEmitter } from "events";`

**Comment by saul** [comment]
Thanks for the report. Out of interest what version of node/npm are you using?

**Comment by agokhale1** [comment]
I'm using node 12.16.1, npm 6.9.2, and TypeScript 3.9.5

**Comment by markbmullins** [comment]
I am also having typescript compilation error.

It cannot find the following type in about 110 places.

```
node_modules/demofile/dist/demo.d.ts:112:44 - error TS2304: Cannot find name 'RequiredNonNullable'.

112     on(message: "net_NOP", listener: (msg: RequiredNonNullable<ICNETMsg_NOP>) => void): this;  
```

**Comment by bureus** [comment]
@markbmullins I have the same issue, did you get any idea of how to resolve it?

**Comment by saul** [comment]
I've just fixed this problem and added a test to make sure we don't regress.

I'll try to get this published as v1.9.0 asap

**Comment by saul** [comment]
v1.9.0 is now released
